In [63]:
import numpy as np
import pandas as pd

In [13]:
data = pd.read_csv("philadelphia_2019.csv")
weatherData = pd.read_csv("weather_hourly_philadelphia.csv")

# quick overview

In [62]:
data.head()

start_time             end_time  start_station_id  end_station_id  \
0  2019-01-01 00:19:00  2019-01-01 00:27:00              3049            3007   
1  2019-01-01 00:30:00  2019-01-01 00:37:00              3005            3007   
2  2019-01-01 00:52:00  2019-01-01 01:05:00              3166            3169   
3  2019-01-01 00:55:00  2019-01-01 01:04:00              3058            3103   
4  2019-01-01 01:05:00  2019-01-01 01:17:00              3182            3028   

   bike_id user_type    start_station_name  \
0    14495  Indego30       Foglietta Plaza   
1     5332  Day Pass     Welcome Park, NPS   
2    14623  Indego30  Frankford & Belgrade   
3    11706  Indego30      20th & Fairmount   
4    11039  Indego30         17th & Sansom   

                            end_station_name  
0                     11th & Pine, Kahn Park  
1                     11th & Pine, Kahn Park  
2                                 2nd & Race  
3  27th & Master, Athletic Recreation Center  
4                           4th & Bainbridge

In [48]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 744260 entries, 0 to 744259
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   start_time          744260 non-null  object
 1   end_time            744260 non-null  object
 2   start_station_id    744260 non-null  int64 
 3   end_station_id      744260 non-null  int64 
 4   bike_id             744260 non-null  int64 
 5   user_type           744225 non-null  object
 6   start_station_name  744260 non-null  object
 7   end_station_name    744260 non-null  object
dtypes: int64(3), object(5)
memory usage: 45.4+ MB


# dealing with missing values

In [53]:
data.isnull().sum()

start_time             0
end_time               0
start_station_id       0
end_station_id         0
bike_id                0
user_type             35
start_station_name     0
end_station_name       0
dtype: int64

because there are only 35 out of 744260 rows with missing values, we can simply drop these rows

In [64]:
data_clean = data.dropna(axis=0, inplace=False)
data_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 744225 entries, 0 to 744259
Data columns (total 8 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   start_time          744225 non-null  object
 1   end_time            744225 non-null  object
 2   start_station_id    744225 non-null  int64 
 3   end_station_id      744225 non-null  int64 
 4   bike_id             744225 non-null  int64 
 5   user_type           744225 non-null  object
 6   start_station_name  744225 non-null  object
 7   end_station_name    744225 non-null  object
dtypes: int64(3), object(5)
memory usage: 51.1+ MB
